core/dbt/events/functions.py

```diff
@@ -114,6 +110,16 @@ def cleanup_event_logger():
     EVENT_MANAGER.callbacks.clear()
 
 
+# The default event manager will not log anything, but some tests run code that
+# generates events, without configuring the event manager, so we create an empty
+# manager here until there is a better testing strategy in place.
+EVENT_MANAGER: EventManager = EventManager()
```

Comment:
I noticed a functional regression related to the fact that `EVENT_MANAGER` does not have any logs configured, prior to `setup_event_logger` being called. If dbt runs into configuration errors during initialization, they are not surfaced to the user: https://github.com/dbt-labs/dbt-core/issues/6434

Reply:
How would that work in dbt-rpc?

Reply:
IMO only stdout logging is acceptable for now. My sense is, we really only expect these logs to fire if something goes wrong during initialization. It's okay to have some hardcoded, generic, unconfigurable behavior that just ensures we're showing the error to the user.

We'll need to figure out the story for `dbt-server`, since I think the initial plan is to tail the `logs/dbt.log` file. If we think the longer-term answer is registering a custom callback, that might solve our problem.

As far as dbt-rpc, all we need is ongoing compatibility with the legacy logger for v1.4. The way I understand the code in this PR, if the `ENABLE_LEGACY_LOGGER` env var is enabled, we'll immediately flip on the legacy logger, without waiting for the call to `setup_event_logger`.

Reply:
@jtcohen6 Your understanding is correct w.r.t. dbt-rpc

From speaking to @racheldaniel, I have a fairly good idea how dbt-server is using logging now. They do their own configuration of the EVENT_MANAGER before running tasks, so I feel confident that we have a good story for them.

I'm going to prepare a PR to address #6434, now. I believe that will restore all of the important logging behaviors we had in 1.3, and I have to apologize for not considering how important it was to get something set up before the call to setup_event_logger() is made.

With all that said, there are some interesting thing to note about the file logging behavior of the legacy logger. I found earlier this week that it starts recording messages after it is set up, which is immediately dbt starts. It then writes the recorded messages to the log file (if any) when the correct file path has been determined, and stops recording. My hope is that in 1.5 we can find a way to start all logging, both console and file, immediately when dbt starts. Does that seem feasible?

Reply:
> I have to apologize for not considering how important it was to get something set up before the call to setup_event_logger() is made.

I'm glad we caught it early! I think the failure here was lack of sufficient inline code comment around the event system initialization, to scare you away from what was a perfectly sensible refactor. "We need to configure these defaults here, even though it looks like we shouldn't, because some log events _might_ arrive before we have the real inputs for proper configuration — namely, if we encounter an early exception."

> I found earlier this week that it starts recording messages after it is set up, which is immediately dbt starts. It then writes the recorded messages to the log file (if any) when the correct file path has been determined, and stops recording.

IIRC, it does this by storing logs in a "buffer," and then "replaying" the buffer once the proper log path has been determined. A little too magical for my taste buds, but this might be an acceptable pattern to emulate.

https://github.com/dbt-labs/dbt-core/blob/5e4e917de5f4da7b24862a2d9b14a2709686e3ff/core/dbt/logger.py#L417-L421

> My hope is that in 1.5 we can find a way to start all logging, both console and file, immediately when dbt starts. Does that seem feasible?

Maybe! We could roll our own similar version of a stateful "buffer"? Or split `log-path` out of `dbt_project.yml`, so that we can know where we're logging without having to parse any config/files? (Feels related: #6207)

FWIW in the happy path, when config initialization succeeds (99% of the time), I don't think we're firing any log events before `setup_event_logger` is called.